infra/resources/_modules/storage_audit/data.tf

```diff
@@ -0,0 +1,6 @@
+data "azurerm_private_dns_zone" "privatelink_blob_core_windows_net" {
```

Comment:
I suggest to keep `data` outside of the inner modules in order to reuse them and avoid dependencies